DataFormats/Common/interface/RandomNumberGeneratorState.h

```diff
@@ -0,0 +1,26 @@
+#ifndef DataFormats_Common_RandomNumberGeneratorState_h
+#define DataFormats_Common_RandomNumberGeneratorState_h
+
+/*----------------------------------------------------------------------
+  
+RandomNumberGeneratorState is used to communicate with an external process
+----------------------------------------------------------------------*/
+
+#include <vector>
+namespace edm {
+  struct RandomNumberGeneratorState {
+    RandomNumberGeneratorState() = default;
+    RandomNumberGeneratorState(std::vector<unsigned long> iState, long iSeed)
+        : state_(std::move(iState)), seed_{iSeed} {}
+
+    RandomNumberGeneratorState(RandomNumberGeneratorState const&) = default;
+    RandomNumberGeneratorState(RandomNumberGeneratorState&&) = default;
+
+    RandomNumberGeneratorState& operator=(RandomNumberGeneratorState const&) = default;
+    RandomNumberGeneratorState& operator=(RandomNumberGeneratorState&&) = default;
+
+    std::vector<unsigned long> state_;
+    long seed_;
```

Comment:
Would encapsulation of the member data make sense here?

Reply:
This isn't a 'real' data format since it never is put in a Principal. It is really an internal detail of how the processes communicate. I put it here since I needed a ROOT dictionary to be generated and this class will be used by the FWCore test as well as the upcoming generator code. That is why I just used a struct.